pkg/traces/servicegraphprocessor/processor.go

```diff
@@ -247,11 +280,15 @@ func (p *processor) consume(trace pdata.Traces) error {
 
 			for k := 0; k < ils.Spans().Len(); k++ {
 
-				if p.store.ItemCount() >= p.maxItems {
-					remainingSpans := float64(ils.Spans().Len() - k)
-					p.serviceGraphDroppedSpansTotal.WithLabelValues(svc.StringVal()).Add(remainingSpans)
+				// Check if the store can process the remaining spans
+				if p.store.len() >= p.maxItems {
+					// todo: try to evict expired items before dropping more
+					droppedSpans := ils.Spans().Len() - k
```

Comment:
i think this will now double count spans b/c droppedSpans includes all future spans in `ils.Spans()` and you're looping all the way to the end.

also you don't really know for sure that you can't store the incoming edge b/c you haven't checked yet. at this point adding the span may simply match it up with another edge that already exists in the store or it might not even really be an edge span at all.

maybe `upsertEdge` should return errTooManyItems and then you can just: `totalDroppedSpans++`

Reply:
Good idea. The dropped spans logic has been broken all this time 🤦‍♂️ .

It makes more sense to move `maxItems` to the store, than having the processor handle it.